### PR TITLE
[IR] Replace uses of IRBuilder::getInt8PtrTy with getPtrTy. NFC

### PR DESCRIPTION
**Pull request by topperc**

### CODE REVIEW

**Review by aeubanks — APPROVED** [review]

**Review by MaskRay — APPROVED** [review]

File: llvm/lib/Transforms/Utils/SanitizerStats.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ StructType *SanitizerStatReport::makeModuleStatsTy() {
 void SanitizerStatReport::create(IRBuilder<> &B, SanitizerStatKind SK) {
   Function *F = B.GetInsertBlock()->getParent();
   Module *M = F->getParent();
-  PointerType *Int8PtrTy = B.getInt8PtrTy();
+  PointerType *Int8PtrTy = B.getPtrTy();
```
Comment (MaskRay) [inline llvm/lib/Transforms/Utils/SanitizerStats.cpp:45]: Rename the variable

Comment (MaskRay) [reply llvm/lib/Transforms/Utils/SanitizerStats.cpp:45]: Or remove it if only used once

File: llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp [hunk 2]
```diff
@@ -1806,7 +1806,7 @@ OpenMPIRBuilder::createTask(const LocationDescription &Loc,
         ++P;
       }
 
-      DepArrayPtr = Builder.CreateBitCast(DepArray, Builder.getInt8PtrTy());
+      DepArrayPtr = Builder.CreateBitCast(DepArray, Builder.getPtrTy());
```
Comment (nikic) [inline llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp:1809]: Can remove this call entirely.

File: llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp [hunk 3]
```diff
@@ -2104,7 +2104,7 @@ OpenMPIRBuilder::InsertPointTy OpenMPIRBuilder::createReductions(
   Function *Func = Builder.GetInsertBlock()->getParent();
   Module *Module = Func->getParent();
   Value *RedArrayPtr =
-      Builder.CreateBitCast(RedArray, Builder.getInt8PtrTy(), "red.array.ptr");
+      Builder.CreateBitCast(RedArray, Builder.getPtrTy(), "red.array.ptr");
```
Comment (nikic) [inline llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp:2107]: This one as well.

File: llvm/lib/Transforms/Coroutines/CoroSplit.cpp [hunk 4]
```diff
@@ -1481,7 +1481,7 @@ static void handleNoSuspendCoroutine(coro::Shape &Shape) {
       IRBuilder<> Builder(AllocInst);
       auto *Frame = Builder.CreateAlloca(Shape.FrameTy);
       Frame->setAlignment(Shape.FrameAlign);
-      auto *VFrame = Builder.CreateBitCast(Frame, Builder.getInt8PtrTy());
+      auto *VFrame = Builder.CreateBitCast(Frame, Builder.getPtrTy());
```
Comment (nikic) [inline llvm/lib/Transforms/Coroutines/CoroSplit.cpp:1484]: This one as well.

File: llvm/lib/Transforms/Utils/AMDGPUEmitPrintf.cpp [hunk 5]
```diff
@@ -165,7 +165,7 @@ static Value *callAppendStringN(IRBuilder<> &Builder, Value *Desc, Value *Str,
 static Value *appendString(IRBuilder<> &Builder, Value *Desc, Value *Arg,
                            bool IsLast) {
   Arg = Builder.CreateBitCast(
-      Arg, Builder.getInt8PtrTy(Arg->getType()->getPointerAddressSpace()));
+      Arg, Builder.getPtrTy(Arg->getType()->getPointerAddressSpace()));
```
Comment (nikic) [inline llvm/lib/Transforms/Utils/AMDGPUEmitPrintf.cpp:168]: This one as well.

File: llvm/lib/Transforms/Utils/AMDGPUEmitPrintf.cpp [hunk 6]
```diff
@@ -300,7 +300,7 @@ static Value *callBufferedPrintfStart(
 
   Type *Tys_alloc[1] = {Builder.getInt32Ty()};
   Type *I8Ptr =
```
Comment (nikic) [inline llvm/lib/Transforms/Utils/AMDGPUEmitPrintf.cpp:302]: Rename variable.

File: llvm/lib/Transforms/Utils/RelLookupTableConverter.cpp [hunk 7]
```diff
@@ -160,7 +160,7 @@ static void convertToRelLookupTable(GlobalVariable &LookupTable) {
                                      "reltable.intrinsic");
 
   // Create a bitcast instruction if necessary.
-  if (Load->getType() != Builder.getInt8PtrTy())
+  if (Load->getType() != Builder.getPtrTy())
```
Comment (nikic) [inline llvm/lib/Transforms/Utils/RelLookupTableConverter.cpp:163]: Can remove this if and bitcast.

File: llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp (hunk 2 above)
Comment (bjope) [reply llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp:1809]: `DepArrayPtr = DepArrays; ` would still be needed afaict (so not just remove)

Comment (topperc) [reply llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp:1809]: Yeah. I'm going to rename DepArrayPtr and change the DepArray declaration in this if to an assignment to DepArray declared before the.

File: llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp [hunk 8]
```diff
@@ -2094,7 +2092,7 @@ OpenMPIRBuilder::InsertPointTy OpenMPIRBuilder::createReductions(
     Value *RedArrayElemPtr = Builder.CreateConstInBoundsGEP2_64(
         RedArrayTy, RedArray, 0, Index, "red.array.elem." + Twine(Index));
     Value *Casted =
-        Builder.CreateBitCast(RI.PrivateVariable, Builder.getInt8PtrTy(),
+        Builder.CreateBitCast(RI.PrivateVariable, Builder.getPtrTy(),
```
Comment (nikic) [inline llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp:2095]: One leftover bitcast here.

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**